src/Features/Core/Portable/ChangeSignature/ChangeSignatureTelemetryLogger.cs

```diff
@@ -0,0 +1,213 @@
+
+            // Which transformations were done
+            CommittedSessionAddedRemovedReordered,
+            CommittedSessionAddedRemovedOnly,
+            CommittedSessionAddedReorderedOnly,
+            CommittedSessionRemovedReorderedOnly,
+            CommittedSessionAddedOnly,
+            CommittedSessionRemovedOnly,
+            CommittedSessionReorderedOnly,
+
+            // Signature change specification details
+            CommittedSession_OriginalParameterCount,
+            CommittedSessionWithRemoved_NumberRemoved,
+            CommittedSessionWithAdded_NumberAdded,
+
+            // Signature change commit information
+            CommittedSessionNumberOfDeclarationsUpdated,
+            CommittedSessionNumberOfCallSitesUpdated,
+            CommittedSessionCommitElapsedMS,
+
+            // Added parameter binds or doesn't bind
+            AddedParameterTypeBinds,
+
+            // Added parameter required or optional w/default
+            AddedParameterRequired,
+
+            // Added parameter callsite value options
+            AddedParameterValueExplicit,
+            AddedParameterValueExplicitNamed,
+            AddedParameterValueTODO,
+            AddedParameterValueOmitted
+        }
+
+        internal static void LogChangeSignatureDialogLaunched() =>
+            s_logAggregator.IncreaseCount((int)ActionInfo.ChangeSignatureDialogLaunched);
+
+        internal static void LogChangeSignatureDialogCommitted() =>
+            s_logAggregator.IncreaseCount((int)ActionInfo.ChangeSignatureDialogCommitted);
+
+        internal static void LogAddParameterDialogLaunched() =>
+            s_logAggregator.IncreaseCount((int)ActionInfo.AddParameterDialogLaunched);
+
+        internal static void LogAddParameterDialogCommitted() =>
+            s_logAggregator.IncreaseCount((int)ActionInfo.AddParameterDialogCommitted);
+
+        internal static void LogTransformationInformation(int numOriginalParameters, int numParametersAdded, int numParametersRemoved, bool anyParametersReordered)
+        {
+            LogTransformationCombination(numParametersAdded > 0, numParametersRemoved > 0, anyParametersReordered);
+
+            s_logAggregator.IncreaseCountBy((int)ActionInfo.CommittedSession_OriginalParameterCount, numOriginalParameters);
+
+            if (numParametersAdded > 0)
+            {
+                s_logAggregator.IncreaseCountBy((int)ActionInfo.CommittedSessionWithAdded_NumberAdded, numParametersAdded);
+            }
+
+            if (numParametersRemoved > 0)
+            {
+                s_logAggregator.IncreaseCountBy((int)ActionInfo.CommittedSessionWithRemoved_NumberRemoved, numParametersRemoved);
+            }
+        }
+
+        private static void LogTransformationCombination(bool parametersAdded, bool parametersRemoved, bool parametersReordered)
+        {
+            // All three transformations
+            if (parametersAdded && parametersRemoved && parametersReordered)
```

Comment:
Can we use pattern matching instead of if-statements? it might make this easier to understand

Reply:
How exactly?